.github/workflows/ci.yaml

```diff
@@ -34,3 +34,49 @@ jobs:
 
       - name: Run tests
         run: uv run pytest tests -m 'not integration'
+
+  # This runs integration tests of the OpenAI API (via the Ollama provider).
+  # This is an alternative to clouds, as testing those on PR will leak secrets.
+  ollama:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install UV
+        run: curl -LsSf https://astral.sh/uv/install.sh | sh
+
+      - name: Source Cargo Environment
+        run: source $HOME/.cargo/env
+
+      - name: Set up Python 3.12
+        run: uv python install 3.12
+
+      - name: Cache Ollama models
+        id: cache-ollama
+        uses: actions/cache@v4
+        with: # cache key is based on where OLLAMA_MODEL is defined.
+          path: ~/.ollama/models  # default directory for Ollama models
+          key: ollama-${{ hashFiles('./src/exchange/providers/ollama.py') }}
+
+      - name: Install Ollama
+        run: curl -fsSL https://ollama.com/install.sh | sh
```

Comment:
confusing part is the log here ends up saying "hey I'm running" when well it isn't ;) I used act locally to figure it out.

```
act pull_request -P ubuntu-latest=ghcr.io/catthehacker/ubuntu:act-latest --container-architecture linux/amd64
```